sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharp/CSharpExplodedGraph.cs

```diff
@@ -519,12 +524,15 @@ protected override void VisitInstruction(ExplodedGraphNode node)
                     newProgramState = VisitVarPattern((VarPatternSyntaxWrapper)instruction, newProgramState);
                     break;
 
-                case SyntaxKindEx.RecursivePattern: // https://github.com/SonarSource/sonar-dotnet/issues/2937
                 case SyntaxKindEx.ConstantPattern:
                     // The 0 in 'case 0 when ...'
                     // Do nothing
                     break;
 
+                case SyntaxKindEx.RecursivePattern:
+                    VisitRecursivePattern((RecursivePatternSyntaxWrapper)instruction);
+                    break;
+
```

Comment:
`VisitRecursivePattern` should not appear for this instruction: the goal here is to handle the recusrive pattern the same way `DeclarationPattern` is currently handled for `IsPattern`. That means that we ignore the instruction itself (for `DeclarationPattern` it does not appear as instruction in CFG), and the pattern is handled at the higher level: inside `IsPatternExpression` (as updated above), and inside `SwitchExpressionArm` for switch

Reply:
Good point. I've removed the call from here and updated the `VisitSwitchExpressionArmBinaryBranch` to call `VisitRecursivePattern`.
It does nothing for now but at least it will make sure that we will not update just a case and forget about the other.